django_fido/backends.py

```diff
@@ -56,24 +85,36 @@ def authenticate(self, request: HttpRequest, user: AbstractBaseUser, fido2_serve
             raise PermissionDenied("Counter didn't increase.")
         return user
 
-    def mark_device_used(self, device, counter):
-        """Update FIDO 2 device usage information."""
-        if counter == 0 and device.counter == 0:
-            # Counter is unsupported by the device, bail out early
-            return
-        if counter <= device.counter:
-            _LOGGER.info("FIDO 2 authentication failed because of not increasing counter.")
-            raise ValueError("Counter didn't increase.")
-        device.counter = counter
-        device.full_clean()
-        device.save()
 
-    def get_user(self, user_id):
-        """Return user based on its ID."""
+class Fido2PasswordlessAuthenticationBackend(BaseFido2AuthenticationBackend):
+    """Authenticate user using FIDO 2 passwordlessly using supplied user handle."""
+
+    def authenticate(self, request: HttpRequest, user: Optional[AbstractBaseUser], fido2_server: Fido2Server,
+                     fido2_state: Dict[str, bytes], fido2_response: Dict[str, Any]) -> Optional[AbstractBaseUser]:
+        """Authenticate using FIDO 2."""
+        user_handle = base64.b64decode(fido2_response['user_handle']).decode('utf-8')
         try:
-            return get_user_model().objects.get(pk=user_id)
-        except get_user_model().DoesNotExist:
+            authenticator = Authenticator.objects.get(user_handle=user_handle)
```

Comment:
This can probably throw `DoesNotExist` which should be handled.

Reply:
It does get handled? line 106

Reply:
Ha, probably overlooked that :)